### PR TITLE
Implement automated testing and CI

### PR DESCRIPTION
**Pull request by jb-adams**
* testing with TestNG
* CI with Github Actions

### CODE REVIEW

**Comment by yash-puligundla** [comment]
Hi Jeremy. 
This PR currently shows the number of files changed as 59. I am guessing it's because it has cascaded changes from the branch, feature/issue-2-initial-app. 
Since the previous [PR](https://github.com/ga4gh/ga4gh-testbed-infrastructure/pull/8) is approved and merged, would rebasing the current branch `feature/issue-10-testing-ci` to the latest develop branch show the true files changed?

**Comment by jb-adams** [comment]
> Hi Jeremy. This PR currently shows the number of files changed as 59. I am guessing it's because it has cascaded changes from the branch, feature/issue-2-initial-app. Since the previous [PR](https://github.com/ga4gh/ga4gh-testbed-infrastructure/pull/8) is approved and merged, would rebasing the current branch `feature/issue-10-testing-ci` to the latest develop branch show the true files changed?

Yep, I will do the rebase now

**Comment by jb-adams** [comment]
@yash-puligundla I rebased and restored the tests, the number of edited files has reduced a lot and they are mostly in the `src/test` package

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    - name: Setting up Gradle Wrapper
+      run: "gradle wrapper"
+
+    - name: Validating Wrapper
+      uses: gradle/wrapper-validation-action@v1
+
+    - name: Setting up the SQLite database and Make
+      run: | 
+        sqlite3 -version
```
Comment (yash-puligundla) [inline .github/workflows/tests.yml:26]: Does ubuntu come with pre-installed sqlite? If not, where are we installing sqlite?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Tests
+on: [push, pull_request, workflow_dispatch] #workflow_dispatch works only if its active in the main branch
```
Comment (yash-puligundla) [inline .github/workflows/tests.yml:2]: Does this mean this workflow runs every time there is a push to this branch? Should we expect the readme to be updated with the latest numbers whenever there is a push?

**Comment by yash-puligundla** [comment]
Looks great to me!

**Review by yash-puligundla — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (jb-adams) [reply .github/workflows/tests.yml:26]: yup, I believe it's pre-installed

File: .github/workflows/tests.yml (hunk 2 above)
Comment (jb-adams) [reply .github/workflows/tests.yml:2]: the README will just show the latest coverage badge for the `main` branch. Markdown isn't sophisticated enough to parameterize the badge image URL and render a different badge for different branches. But clicking on the badge links out to the full coverage report across all branches on codecov

**Pull request merged**